tests/req_tests.rs

```diff
@@ -89,3 +89,103 @@ fn test_add_route() {
+    let query: Vec<&str> = ["arg1=1&arg2=2"].to_vec();
+    request = request.params_from_route(query);
+    let compare: Vec<String> = Vec::new();
+    assert_eq!(request.get_params() , compare)
+}
+
+#[test]
+fn test_args_from_route() {
+    let mut request = Request::new().with_route("/test/route/param?arg1=1".to_string());
+    let mut query: Vec<&str> = ["param", "arg1=1"].to_vec();
+    let args: HashMap<String,String> = [("arg1".to_string(), "1".to_string())]
+                                        .iter().cloned().collect();
+    request = request.args_from_route(query.as_mut());
+    assert_eq!(request.get_args() , args)
+}
+
+#[test]
+fn test_multiple_args_from_route() {
+    let mut request = Request::new().with_route("/test/route/param?arg1=1&arg2=2".to_string());
+    let mut query: Vec<&str> = ["param", "arg1=1&arg2=2"].to_vec();
+    let args: HashMap<String,String> = [("arg1".to_string(), "1".to_string()),
+                                        ("arg2".to_string(), "2".to_string())]
+                                        .iter().cloned().collect();
+    request = request.args_from_route(query.as_mut());
+    assert_eq!(request.get_args() , args)
+}
+
+#[test]
+fn test_empty_args_from_route() {
+    let mut request = Request::new().with_route("/test/route/param?".to_string());
+    let mut query: Vec<&str> = ["param"].to_vec();
+    let args: HashMap<String,String> = HashMap::new();
+    request = request.args_from_route(query.as_mut());
+    assert_eq!(request.get_args() , args)
+}
+
+#[test]
+fn test_parse_query_string_single() {
+    let mut request = Request::new().with_route("/test/route/param?arg1=1".to_string());
+    let args: HashMap<String,String> = [("arg1".to_string(), "1".to_string())]
+                                        .iter().cloned().collect();
+    request = request.parse_query_string();
+    assert_eq!(request.get_params() , ["param".to_string()].to_vec());
+    assert_eq!(request.get_args() , args)
+}
+
+#[test]
+fn test_parse_query_string_multiple() {
+    let mut request = Request::new()
+                              .with_route("/test/route/param1?param2?arg1=1&arg2=2".to_string());
+    let args: HashMap<String,String> = [("arg1".to_string(), "1".to_string()),
+                                        ("arg2".to_string(), "2".to_string())]
+                                        .iter().cloned().collect();
+    request = request.parse_query_string();
+    assert_eq!(request.get_params() , ["param1".to_string(), "param2".to_string()].to_vec());
+    assert_eq!(request.get_args() , args)
+}
+
+#[test]
+fn test_parse_query_string_none() {
+    let mut request = Request::new().with_route("/test/route/".to_string());
+    let params: Vec<String> = Vec::new();
+    let args: HashMap<String,String> = HashMap::new();
+    request = request.parse_query_string();
+    assert_eq!(request.get_params() , params);
+    assert_eq!(request.get_args() , args)
+}
+
+#[test]
+fn test_parse_query_string_wildcard() {
```

Comment:
This isn't the correct way to test this. The request will contain a route that does not have the wildcard in it. The wildcard is in the route configuration in the server. The route map might have an entry like this:
```
"GET /home/{}" => my_func
```
and the request's route could look like: `"GET /home/stuff"`. This should route to `my_func` and the request should contain the arg 'stuff'.